lib/teamsnap.rb

```diff
@@ -131,18 +135,51 @@ def init(opts = {})
       apply_endpoints(self, collection) && true
     end
 
-    def run(via, href, args = {})
+    def run(via, href, args = {}, opts = {})
       resp = client.send(via, href, args)
 
       if resp.status == 200
-        Oj.load(resp.body)
-          .fetch(:collection)
+        collection = Oj.load(resp.body).fetch(:collection)
+        if opts.include?(:backup_cache_file)
+          TeamSnap.write_backup_file(opts[:backup_cache_file], collection)
+        end
```

Comment:
tell, don't ask

`write_backup(opts[:backup_cache_file], collection` and make `write_backup` check if there's a cache or not.